### PR TITLE
Bigquery source: test#3

### PR DESCRIPTION
**Pull request by dauresh**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigquery/unit_tests/integration/test_discover.py [hunk 1]
```diff
@@ -89,3 +89,53 @@ def test_simple_discovery(self, http_mocker: HttpMocker) -> None:
         ] == [
             f"{dataset_id}.{table_id}" for dataset_id in dataset_ids for table_id in tables_ids
         ]
+
+        assert not output.errors
+
+    @HttpMocker()
+    def test_no_datasets(self, http_mocker: HttpMocker) -> None:
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-bigquery/unit_tests/integration/test_discover.py:96]: Can you explain what we are trying to test here? Based on [this sheet](https://docs.google.com/spreadsheets/d/1e1n43SrTIomAqrJspSwW4hb-qTaLP6JO4uOBrFjvTDk/edit?gid=1335688273#gid=1335688273) it says `run discover command with valid basic config file with empty dataset_id field`. For me, this would mean that we want to test:
* Given a dataset_id in the config, we only fetch for those
* Given no dataset_id in the config, we fetch for all the datasets

What this test is testing is that if there is no datasets returned by BigQuery, we don't have any stream. This test is a bit surprising because I would guess that if it were the case, we would return a config_error to the user. However, since we can consider this an edge case for now, I'm fine not doing the change, keeping this test as a documentation of the current behavior and add a comment that describes the situation. The same applies for the other test

Comment (dauresh) [reply airbyte-integrations/connectors/source-bigquery/unit_tests/integration/test_discover.py:95]: Check the connector's behaviour when no dataset is found.
The test case you describe is the #25 in the sheet I think ?
